src/components/ImageDepViewer.js

```diff
@@ -0,0 +1,109 @@
+import _ from 'lodash/fp'
+import { Fragment, useEffect, useState } from 'react'
+import { div, h, table, tbody, td, thead, tr } from 'react-hyperscript-helpers'
+import { Ajax } from 'src/libs/ajax'
+
+import { Select } from './common'
+
+
+export const ImageDepViewer = ({ packageLink, namespace }) => {
+  const [tools, setTools] = useState([])
+  const [tool, setTool] = useState('Loading...')
+  const [packageDoc, setPackageDoc] = useState({})
+
+  const packages = _.filter(doc => doc['tool'] === tool, packageDoc)
+
+  const fetchImageDocumentation = async (packageLink, namespace) => {
+    const splitPath = packageLink.split('/')
+    const object = splitPath[splitPath.length - 1]
+    const bucket = splitPath[splitPath.length - 2]
+
+    const file = await Ajax().Buckets.getObjectPreview(bucket, object, namespace, true).then(res => res.json())
+
+    const adaptedDoc = packageDocAdaptor(file)
+    return adaptedDoc
+  }
+
+  useEffect(() => {
+    async function inner() {
+      const docs = await fetchImageDocumentation(packageLink, namespace)
+      setPackageDoc(docs)
+
+      const tools = _.uniq(_.map(doc => doc['tool'], docs))
+      const sortedTools = _.sortBy(tool => tool === 'tools' ? 1 : 0, tools)
+
+      setTools(sortedTools)
+
+      if (!tool) {
+        setTool(sortedTools[0])
+      } else if (!tools.includes(tool)) {
+        setTool(sortedTools[0])
+      } else {
+        //in the case where tool exists and the current tool is in tool, we do not change it if the list of tools has changes so the dropdown does not change
+      }
+    }
+    inner()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
without this, the linter complains about namespace and tools not being included as dependencies, but since the main look interacts with tools, doing this results in an infinite re-render loop.

It seems others have had this issue: https://stackoverflow.com/questions/55840294/how-to-fix-missing-dependency-warning-when-using-useeffect-react-hook